src/sknnr/transformers/_rfnode_transformer.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from typing import Literal
+
+import numpy as np
+from numpy.random import RandomState
+from numpy.typing import NDArray
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.utils.validation import check_is_fitted
+
+from .._base import _validate_data
+
+
+class RFNodeTransformer(TransformerMixin, BaseEstimator):
+    """
+    Transformer to capture node indexes for samples across multiple
+    random forests.
+
+    A random forest is fit to each feature in the training set using
+    `RandomForestRegressor`.  The transformation captures the node indexes
+    for each tree in each forest for each training or target sample.
+
+    This transformer is intended to be used in conjunction with `RFNNRegressor`
+    which captures similarity between node indexes of training and target data
+    and creates predictions using nearest neighbors.
+
+    Parameters
+    ----------
+    n_estimators_per_forest : int, default=50
+        Number of trees in each random forest.
+    max_features : {'sqrt', 'log2', None}, int or float, default='sqrt'
+        Number of features to consider when looking for the best split.
+    random_state : int, RandomState instance or None, default=None
+        Controls the random seed given to each forest.  See
+        `RandomForestRegressor` for more information.
+
+    Attributes
+    ----------
+    n_features_in_ : int
+        Number of features seen during `fit`.
+    feature_names_in_ : ndarray of shape (`n_features_in_`)
+        Names of features seen during fit. Defined only when `X` has feature
+        names that are all strings.
+    rfs_ : list of `RandomForestRegressor`
+        The random forests associated with each feature in `y` during `fit`.
+    """
+
+    def __init__(
+        self,
+        n_estimators_per_forest: int = 50,
+        max_features: Literal["sqrt", "log2"] | int | float | None = "sqrt",
+        random_state: int | RandomState | None = None,
+    ):
+        self.n_estimators_per_forest = n_estimators_per_forest
+        self.max_features = max_features
+        self.random_state = random_state
+
+    def fit(self, X, y=None):
+        _, y = _validate_data(self, X=X, y=y, reset=True, multi_output=True)
+        if y.ndim == 1:
+            y = y.reshape(-1, 1)
+
+        self.rfs_ = [
```

Comment:
We can certainly tackle this separately, but I wonder if we should parallelize this with `joblib` based on `n_jobs`? I did a quick test fitting with SWO Ecoplot, the default 50 trees, and `n_jobs=-1` and the time went from 14.1s as-is to 8.7s when parallelizing this loop and passing `n_jobs` to the RF constructor. Creating the forests serially but passing `n_jobs` to the RF constructor took 9.6s.

Reply:
This sounds like a great idea.  I think it would look something like this, but I'm not nearly as familiar as you are with it:
```
from joblib import Parallel, delayed

self.rfs_ = Parallel(n_jobs=self.n_jobs)(
    delayed(lambda i: RandomForestRegressor(
        n_estimators=self.n_estimators_per_forest,
        max_features=self.max_features,
        random_state=self.random_state,
        min_samples_leaf=5,
        n_jobs=self.n_jobs,
    ).fit(X, y[:, i]))(i)
    for i in range(y.shape[1])
)
```
It did seem to slow down tests.  Does that seem right?

Reply:
Yeah, it's strange.  Obviously in tests, we are instantiating these objects quite a bit and there is presumably overhead associated with setting up `joblib`, but I can't reconcile why this counters the speedups that you saw.  I'm OK pushing this to another PR, although you might be the one doing the heavy lifting!

Reply:
You may be right about the joblib overhead causing the slowdown - hopefully this isn't another mystery threading issue, although I could see some strangeness arising from parallelizing the already parallel fitting of RFs... Worth a closer look, for sure.